control/tap-ctl-spawn.c

```diff
@@ -196,5 +232,15 @@ tap_ctl_spawn(void)
 	if (id < 0)
 		EPRINTF("get_id failed, child %d err %d\n", child, errno);
 
+	/* Put the tapdisk in a cgroup slice (best-effort) */
+	if (!slice) {
+#ifdef TAP_CTL_USE_DEFAULT_CGROUP_SLICE
+		/* No option specified; move it to the default slice */
+		tap_ctl_move_to_cgroup(id, "/");
```

Comment:
@jjd27 I was not thinking of tcp_ctl_close because nothing has been necessarily allocated and attached. I thought that killing the process would have been sufficient but it well be my analysis is very simplistic.

Let's not fail, then, and I will raise another ticket to fix it for later versions (I am seriously worried about
"silent" failures).

Would you just use the return code (I think we agreed -1 is sufficient) to notify the user about errors for different
cases?

Case:
1. blktap compiled without the MACRO defined (see comment about macro): we do not even try cgroups so
nothing to log
2. blktap compiled with the MACRO defined, no slice specified: we try to use the default slice but it fails (let's say
cgroups is not enabled in the kernel), so we log that we are not using cgroups at all
3. blktap compiled with the MACRO defined, slice specified: we log that we could not write in that slice and we are not using cgroups (or, given we are handling errors now, you can decide to try with default, your choice).

My point here is: **tap_ctl_move_to_cgroup** can just log the fact that it fails but it does not know how we are
going to handle that. That's why we need to check the return code and log accordingly in the caller.